xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -657,15 +712,30 @@ void CGUIDialogPVRTimerSettings::AddCondition(
   setting->SetDependencies(deps);
 }
 
-int CGUIDialogPVRTimerSettings::GetDateAsInt(const CDateTime &datetime)
+int CGUIDialogPVRTimerSettings::GetDateAsIndex(const CDateTime &datetime)
 {
-  const CDateTime date(datetime.GetYear(), datetime.GetMonth(), datetime.GetDay(), 0, 0, 0);
   time_t t(0);
-  date.GetAsTime(t);
+  const CDateTime date(datetime.GetYear(), datetime.GetMonth(), datetime.GetDay(), 0, 0, 0);
+  if (date > m_maxDateCovered)
+    (m_maxDateCovered + CDateTimeSpan(1, 0, 0, 0)).GetAsTime(t);
+  else if (date < m_minDateCovered)
+    (m_minDateCovered - CDateTimeSpan(1, 0, 0, 0)).GetAsTime(t);
+  else
+    date.GetAsTime(t);
```

Comment:
IMO, no change is needed to this method at all. All you need for the spinner to work is a unique value for every entry. This is achieved if you just use the numerical time_t value. Whether the time value is in covered range is completely unrelated here. Mapping of values out of range is imo unnecessary and makes things only more complicated.